src/runtime/TabsController.ts

```diff
@@ -0,0 +1,132 @@
+    TAB_LIST: `.${TABS_LIST_CLASSNAME}`,
+    TAB: `.${TAB_CLASSNAME}`,
+    TAB_PANEL: `.${TAB_PANEL_CLASSNAME}`,
+};
+
+export interface ISelectTabByIdOptions {
+    scrollToElement: boolean;
+}
+
+export class TabsController extends EventTarget {
+    private _document: Document;
+
+    constructor(document: Document) {
+        super();
+
+        this._document = document;
+        this._document.addEventListener('click', (event) => {
+            const target = getEventTarget(event) as HTMLElement;
+
+            if (isCustom(event) || !this.isValidTabElement(target)) {
+                return;
+            }
+
+            const tab = this.getTabDataFromHTMLElement(target);
+            if (tab) {
+                this.selectTab(tab, target.dataset.diplodocId!);
+            }
+        });
+    }
+
+    selectTabById(id: string, options?: ISelectTabByIdOptions) {
+        const target = this._document.querySelector(
+            `${Selector.TAB}[${TAB_DATA_ID}="${id}"]`,
+        ) as HTMLElement;
+
+        if (!target || !this.isValidTabElement(target)) {
+            return;
+        }
+
+        const tab = this.getTabDataFromHTMLElement(target);
+        if (tab) {
+            this.selectTab(tab, id);
+        }
+
+        if (options?.scrollToElement) {
+            target.scrollIntoView();
+        }
+    }
+
+    selectTab(tab: Tab, currentTabId?: string) {
+        const {group, key} = tab;
+
+        if (!group) {
+            return;
+        }
+
+        const tabs = this._document.querySelectorAll(
+            `${Selector.TABS}[${GROUP_DATA_KEY}="${group}"] ${Selector.TAB}[${TAB_DATA_KEY}="${key}"]`,
+        );
+
+        let found = 0;
+
+        tabs.forEach((element) => {
+            const htmlElem = element as HTMLElement;
+            if (!this.isValidTabElement(htmlElem) || element.classList.contains(ACTIVE_CLASSNAME)) {
+                return;
+            }
+
+            found++;
+
+            const tab = element;
+            const tabList = tab.parentNode;
+            const tabsContainer = tabList?.parentNode;
+            const allTabs = Array.from(tabsContainer?.querySelectorAll(Selector.TAB) || []);
+            const allPanels = Array.from(tabsContainer?.querySelectorAll(Selector.TAB_PANEL) || []);
+            const targetIndex = allTabs.indexOf(tab);
+
+            allTabs.forEach((tab, i) => {
+                const panel = allPanels[i];
+                const isTargetTab = i === targetIndex;
+
+                tab.classList.toggle(ACTIVE_CLASSNAME, isTargetTab);
+                tab.setAttribute('aria-selected', isTargetTab.toString());
+                tab.setAttribute('tabindex', isTargetTab ? '0' : '-1');
+                panel.classList.toggle(ACTIVE_CLASSNAME, isTargetTab);
+            });
+        });
+
+        if (found > 0) {
+            const eventTab: Tab = group.startsWith(DEFAULT_TABS_GROUP_PREFIX)
+                ? {key: tab.key}
+                : tab;
```

Comment:
If tabs doesn't have group, we will emit the "selected" event but without group. We are hiding a technical inner group.